### ISSUE TITLE
Connection pooler with Custom TLS

### ISSUE THREAD

**Issue by Stolr** [zalando/postgres-operator#2283]
- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.8.2

Helm chart 
```ỳaml
    repoURL: 'https://opensource.zalando.com/postgres-operator/charts/postgres-operator'
    targetRevision: 1.9.0
```

- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** [AWS K8s | GCP ... | Bare Metal K8s]

Bare Metal

- **Are you running Postgres Operator in production?** [yes | no]

Yes

- **Type of issue?** [Bug report, question, feature request, etc.]

Bug report + feature request

- **PostGres Version?**

15

---

Hi ,
Using the last version of operator, custom TLS with  connection pooler does not work properly. Here is the log :
```
+ '[' pooler '=' postgres ]
+ '[' -z /tls/tls.key ]
+ ln /tls/tls.key /etc/ssl/certs/pgbouncer.crt
ln: /etc/ssl/certs/pgbouncer.crt: Cross-device lin
```

Part of the issue is resolved here : https://github.com/zalando/postgres-operator/issues/2201

To make it work , I need to change several things :

In `deployment`  definition :
For `env var` From this :
```yaml
            - name: CONNECTION_POOLER_CLIENT_TLS_CRT
              value: /tls/tls.key
            - name: CONNECTION_POOLER_CLIENT_TLS_KEY
              value: /tls
```
To this :

```yaml
            - name: CONNECTION_POOLER_CLIENT_TLS_CRT
              value: /tls/tls.crt
            - name: CONNECTION_POOLER_CLIENT_TLS_KEY
              value: /tls/tls.key
```
And also using image  `registry.opensource.zalan.do/acid/pgbouncer:master-27` instead of `registry.opensource.zalan.do/acid/pgbouncer:master-26`

After doing that , I get the following logs :

```
2023-04-08 18:34:10.457 UTC [1] ERROR TLS setup failed: failed to load certificate file "/etc/ssl/certs/pgbouncer.crt": Permission denied
2023-04-08 18:34:10.457 UTC [1] FATAL TLS setup failed
```
So I also need to change `defaultmode` of the mounted secret 
```yaml
      volumes:
        - name: tls
          secret:
            secretName: pg-certs
            defaultMode: 0777
 ```
( Which is not very clean )

So after theses changes, everything is doing fine.

But of course, the operator tries to revert my change. It works fine because  the new pod keeps crashing , so  my old pods keep running , but having permanently a pod in `CrashLoopBackOff` state. 

Can you please provide a new version  so we  would be able to use our custom certs for the pooler ?

Do you have any idea for a Quick Fix maybe to be able to have a clean state until that ?

Thanks a lot for the great work !

**Comment by FxKu** [comment]
Hello. Could you repeat you test with image registry.opensource.zalan.do/acid/postgres-operator-test:c9f4e9c-pr-2255-3 from PR #2255 ? It should fix the problems described here. Make sure to also define spiloFSGroup in the Postgres manifest or global config. I also face the same errors you described here while testing. But it should be fixed with #2255. The PR is just one e2e test away from getting merged.

**Comment by Stolr** [comment]
Fixed using operator 1.10 !

Thanks !
